karapace/config.py

```diff
@@ -186,6 +187,23 @@ def set_config_defaults(config: ConfigDefaults) -> Config:
     new_config.setdefault("tags", {})
     new_config["tags"]["app"] = "Karapace"
 
+
+    # Set the aiohttp client max size if REST Proxy is enabled and producer max request configuration is altered from default
+    # and aiohttp client max size is not set
+    if new_config["karapace_rest"] and new_config["producer_max_request_size"] > DEFAULT_PRODUCER_MAX_REQUEST and new_config["http_request_max_size"] is None:
+        # REST Proxy API configuration for producer max request size must be taken into account
+        # also for the aiohttp.web.Application client max size.
+        # Always add the aiohttp default client max size as the headroom above the producer max request size.
+        # The input JSON size for REST Proxy is not easy to estimate, lot of small records in single request has
+        # a lot of overhead due to JSON structure.
+        new_config["http_request_max_size"] = new_config["producer_max_request_size"] + DEFAULT_AIOHTTP_CLIENT_MAX_SIZE
+    elif new_config["http_request_max_size"] is None:
+        # Set the default aiohttp client max size
+        new_config["http_request_max_size"] = DEFAULT_AIOHTTP_CLIENT_MAX_SIZE
+    else:
+        # Use the aiohttp client max size from the configuration without altering.
+        pass
```

Comment:
```suggestion
```

suggestion (non-blocking): Skip unnecessary branch.

Reply:
I can remove, the reason why I added `pass` is to be explicit with the use of the value in the configuration.